### DISCUSSION TITLE
Feature Request: Built-in adblocker

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#888] [Ideas]
### What feature would you like to see?

title

### What platform are you seeing the problem on?

Windows

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/162_

**Comment by SomeDudeBrosky** [comment]
Specifically uBlock Origin, as it is the most reliable and accurate

**Comment by ahgraber** [comment]
I wouldn't mind bundling uBlock Origin, but I don't think it makes sense to try to "build in" an adblocker...

**Comment by nghiaxsm** [comment]
Is there any significant benefit from buitlt-in adblocker?

**Reply by Reapimus** [reply]
depending on how a built-in one is designed, it might be way faster and more performant than an extension could ever hope to achieve.

**Reply by youwen5** [reply]
> depending on how a built-in one is designed, it might be way faster and more performant than an extension could ever hope to achieve.

why though? so it can go from imperceptible (ub0) to even more imperceptible by being builtin? you're hitting massive diminishing returns and you take on the overhead of having to maintain your own ad blocker when a gold standard already exists.

plus, client side content blockers can only do so much and ub0 is damn good. really the only substantial improvement is to implement filtering at the resolver level like pi-hole and adguard dns, which is out of scope for a browser.

**Reply by JSGRD22** [reply]
Perhaps a built-in blocker could add something new by having the default configurations include blocklists that are disabled by default in a newly-installed extension but that most people would want to use (by built-in I am referring to simply bundling uBO, I don't think there's anything else worth doing)

**Reply by xe-3** [reply]
> Perhaps a built-in blocker could add something new by having the default configurations include blocklists that are disabled by default in a newly-installed extension but that most people would want to use (by built-in I am referring to simply bundling uBO, I don't think there's anything else worth doing)

I'm pretty sure (but not 110% certain) that bundling uBO would allow the option of pre-configuring custom lists. But really, the default lists of uBO are probably optimal for most people, and are by far the most tested and quicker to be fixed/updated when there are issues. Diverging from the default lists should be done only when you have specific reasons to do so in most cases.

**Comment by SomeDudeBrosky** [comment]
> Is there any significant benefit from buitlt-in adblocker?

More users would have an ad free experience, as not everyone even knows they exist, eg. giving more users a better experience.

**Reply by youwen5** [reply]
it's probably more sane to just bundle ublock origin like librewolf does, i dont think zen needs to waste resources on an integrated blocker at this time when ub0 exists and firefox isnt threatened by manifest v3 like chromium

**Comment by ShayBox** [comment]
I have heard that Brave's AdBlock which is written in Rust is fairly performant, but I have no idea how it really compares to the gold standard uBlock Origin.

If it was based on anything else most people would just disable it and use uBlock Origin instead, so it would probably feel like a waste of time.

EDIT: Side issue, but uBlock Origin doesn't seem to work in the sidebar, so a built-in option would hopefully work with this too.

**Reply by xe-3** [reply]
> I have heard that Brave's AdBlock which is written in Rust is fairly performant, but I have no idea how it really compares to the gold standard uBlock Origin.

According to some of Brave's own developers, their builtin adblocker (which is inspired by and borrows from uBO) is more performant, but at levels that will be imperceptible to humans (think micro-seconds) and won't noticeably impact the browsing experience.

Brave's CEO has also stated in the past that:
> "[uBlock Origin is] a really strong model of the kind of mentality and approach the space should bring to user privacy"

and

> "uBO does more by default, and has more knobs for its users, than Brave shields has or will likely ever have."

In any case, using an adblocker will [generally speed up browsing](https://x.com/gorhill/status/1792648742752981086/photo/1) compared to not using an adblocker because all of the crap it keeps from loading, so it should be a step in the right direction whichever (reputable and well established) option is chosen.

Personally I have a strong preference for uBO.

**Reply by artem-nefedov** [reply]
I switched to Brave specifically because uBlock was not doing a good job on certain platforms.
(Brave defaults also weren't good enough, but it does provide menu options to solve the problem)

**Comment by antikovt** [comment]
If there are any potential benefits to be had when bundling uBO, then sure, but I find it a little hard to believe. Otherwise perhaps a "Recommended Extensions" page during onboarding?

**Comment by Axenide** [comment]
Maybe... But I think it would be better (and easier) to recommend uBlock Origin when setting up the browser, like an option that says "Install uBlock Origin to block ads and tracking?" or even a few recommended extensions, like a multi-selection menu with, for example, DuckDuckGo Privacy Essentials, uBlock Origin, etc.

**Reply by swagato-c** [reply]
I dont think just UBO would solve the problem. Some sites like Youtube can detect it as an addon and prevent you from watching stuff. Whereas the the UBO integration in brave is good enough for Youtube to keep on guessing.

**Reply by JSGRD22** [reply]
uBlock origin is sufficiently powerful to bypass youtube's anti-adblock system. Besides, brave does not use uBo.

**Comment by bloor** [comment]
Just make uBlock origin extension/mod. 
And pre-install it as a bundle. OR EVEN BETTER, just make it a checkbox during install.
Anyone "annoyed" can remove it.

**Comment by iandesignandbuild** [comment]
Really like Zen , but I need something like uBlock origin added. Whole point for me is using the internet in a calm neurological state so i can concentrate properly.

Pollution has all but been eliminated on firefox with ublock. It's an essential requirement before I can become a user.
Thanks for all the hard work so far, 
Ian.

**Reply by youwen5** [reply]
> Really like Zen , but I need something like uBlock origin added. Whole point for me is using the internet in a calm neurological state so i can concentrate properly.
> 
> 
> 
> Pollution has all but been eliminated on firefox with ublock. It's an essential requirement before I can become a user.
> 
> Thanks for all the hard work so far, 
> 
> Ian.

?

You can just install it yourself using the firefox extensions store.

**Reply by Axenide** [reply]
Why don't you just... [Install it](https://addons.mozilla.org/es-ES/firefox/addon/ublock-origin/)?

**Comment by iandesignandbuild** [comment]
Than ks very much! - I'll try again with that link.
 I'm still learning to navigate Zen -  and the 'plug-ins' just were not coming up anywhere I could open, I presumed they must be unavailable.  If I just get to spend more time using it I am sure I'll work things out for myself well enough.
Ian.

-- 
 Secured with Tuta Mail: 
 https://tuta.com/free-email


13 Mar 2025, 20:16 by ***@***.***:

**Comment by iandesignandbuild** [comment]
Ha! That was so easy. And ,I am seeing another point of workspaces now - add-ons/plug-ins installing only in the active workspace, not the others - nice.Thanks again,
Ian.
-- 
 Secured with Tuta Mail: 
 https://tuta.com/free-email


Mar 14, 2025, 12:09 by ***@***.***:

**Comment by ahgraber** [comment]
Not to necro this thread, but waterfox just shared they'll be integrating Brave's adblock (and the rationale): https://www.waterfox.com/blog/15-years-of-forking/#what-waterfox-is-in-2026

**Reply by mr-cheffy** [reply]
We'll see how they are doing it, we may also start doing something similar if its feasible.

**Reply by Zebra2711** [reply]
@mr-cheffy Waterfox has it now, so it would be good if Zen had it too

**Reply by mr-cheffy** [reply]
Seems like firefox added their own brave adblock implementation at https://bugzilla.mozilla.org/show_bug.cgi?id=2013888, we'll see what they do with it now

**Reply by Dr-Sauce** [reply]
Any updates/plans on this?
